File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

vector<int> solution(int k, vector<int> score) 
{
    int day = score.size();
    
    vector<int> vec;
    vector<int> result;
    for(int i=0; i<day; i++)
    {
        if(vec.size() < k)
        {
            vec.push_back(score[i]);
            sort(vec.begin(), vec.end());
            result.push_back(vec[0]);
            continue;
        }
        
        if(vec.size() == k)
        {
            if(vec[0] < score[i])
            {
                vec[0] = score[i];
                sort(vec.begin(), vec.end());
                result.push_back(vec[0]);
            }
            else
            {
                result.push_back(vec[0]);
            }
        }       
    }
    return result;  
}

// for(int &i : vec)
// {
//     cout << i;
// }

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cs
using System;
using System.Linq;
using System.Collections.Generic;

public class Solution 
{
    public int[] solution(int k, int[] score) 
    {
        List<int> list = new List<int>();
        int[] result = new int[score.Length];
        
        for(int i=0; i<score.Length; i++)
        {
            if(list.Count < k)
            {
                list.Add(score[i]);
            }
            else if(list.Count == k)
            {
                if(list[0] < score[i])
                {
                    list[0] = score[i];
                }
            }
            list = list.OrderBy(x => x).ToList();
            result[i] = list[0];
        }
        return result;
    }
}